src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs

```diff
@@ -550,10 +555,13 @@ public async Task Match(RegexEngine engine, string pattern, string input, RegexO
             {
                 // Use Match(string) or Match(string, string, RegexOptions)
                 VerifyMatch(r.Match(input));
-                VerifyMatch(Regex.Match(input, pattern, options));
+
+                // Make sure to pass the NonBacktracking option to the static Match method
+                RegexOptions engine_with_options = engine == RegexEngine.NonBacktracking ? options | RegexHelpers.RegexOptionNonBacktracking : options;
+                VerifyMatch(Regex.Match(input, pattern, engine_with_options));
 
                 Assert.Equal(expectedSuccess, r.IsMatch(input));
-                Assert.Equal(expectedSuccess, Regex.IsMatch(input, pattern, options));
+                Assert.Equal(expectedSuccess, Regex.IsMatch(input, pattern, engine_with_options));
```

Comment:
This isn't just an issue with NonBacktracking; RegexOptions.Compiled also isn't being factored in.  And these statics don't make sense for the source generator(s).

I think we should address it instead like:
https://github.com/dotnet/runtimelab/pull/1628